src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -134,6 +133,22 @@ export class FocusMonitor implements OnDestroy {
     this._windowFocusTimeoutId = setTimeout(() => this._windowFocused = false);
   }
 
+  /**
+   * Event listener for `focus` and 'blur' events on the document.
+   * Needs to be an arrow function in order to preserve the context when it gets bound.
+   */
+  private _documentFocusAndBlurListener = (event: FocusEvent) => {
+    const target = event.target as HTMLElement|null;
+    // We need to walk up the ancestor chain in order to support `checkChildren`.
+    for (let el = target; el; el = el.parentElement) {
```

Comment:
Couldn't we use `parent.contains(child)` here? I've tried benchmarking it before and marginally faster.

Reply:
No. We're not looking for the one element to call onFocus/onBlur on - they may be called on multiple ones.